grpc/src/main/java/com/linecorp/armeria/internal/server/grpc/AbstractServerCall.java

```diff
@@ -339,24 +336,39 @@ public void onRequestMessage(DeframedMessage message, boolean endOfStream) {
                 }
             }
 
-            final boolean grpcWebText = GrpcSerializationFormats.isGrpcWebText(serializationFormat);
-            request = marshaller.deserializeRequest(message, grpcWebText);
-            maybeLogRequestContent(request);
-
-            if (unsafeWrapRequestBuffers && buf != null && !grpcWebText) {
-                GrpcUnsafeBufferUtil.storeBuffer(buf, request, ctx);
-            }
-
             if (blockingExecutor != null) {
-                blockingExecutor.execute(() -> invokeOnMessage(request, endOfStream));
+                blockingExecutor.execute(() -> deserializeAndInvoke(message, endOfStream));
```

Comment:
If we run the deserialization in the blocking executor, what happens to the thrown deserialization and size limit exceptions?